doc/overview/roadmap.rst

```diff
@@ -0,0 +1,170 @@
+.. _roadmap:
+
+Roadmap
+=======
+
+This page describes some of the major medium- to long-term goals for
+MNE-Python. These are goals that require substantial effort and/or
+API design considerations. Some of these may be suitable for Google Summer of
+Code projects, while others require more extensive work.
+
+.. contents:: Page contents
+   :local:
+   :depth: 1
+
+
+Clustering statistics API
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The current clustering statistics code has limited functionality. It should be
+re-worked to create a new ``cluster_based_statistic`` or similar function.
+In particular, the new API should:
+
+1. Support mixed within- and between-subjects designs, different statistical
+   functions, etc. This should be done via a ``design`` argument that mirrors
+   :func:`patsy.dmatrices` or similar community standard (e.g., this is what
+   is used by :class:`statsmodels.regression.linear_model.OLS`).
+2. Have clear tutorials showing how different contrasts can be done (toy data).
+3. Have clear tutorials showing some common analyses on real data (time-freq,
+   sensor space, source space, etc.)
+4. Not introduce any significant speed penalty (e.g., < 10% slower) compared
+   to the existing, more specialized/limited functions.
+
+
+3D visualization
+^^^^^^^^^^^^^^^^
+
+Historically we have used Mayavi for 3D visualization, but have faced
+limitations and challenges with it. We should work to use some other backend
+(e.g., PyVista) to get major improvements, such as:
+
+1. Proper notebook support (XTK through pyvtk?)
```

Comment:
```suggestion
1. Proper notebook support (through vtkjs)
```

I don't have a clear idea what vtkjs option will win the war of standard in the community